### PR TITLE
feat(AsyncPipe): should work in detached mode

### LINKED ISSUE #12356: AsyncPipe doesn't update when changeDetectorRef.detach()
**I'm submitting a ...**  (check one with "x")

```
[x] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
`AsyncPipe` doesn't update when `ChangeDetectorRef` for the given component is detached.

<hr>

Now, I'm not sure if this is a bug / feature (probably, very likely a feature).
By looking at [the way change detection is implemented](https://github.com/angular/angular/blob/master/modules/%40angular/common/src/pipes/async_pipe.ts#L137) in `AsyncPipe`, it seems like `this._ref.markForCheck()` doesn't run if `ChangeDetectorRef` is detached (as is also confirmed in [here](https://github.com/angular/angular/issues/7311#issuecomment-189292079)).

The motivation for this is to have full control over when the change detection occurs (by using `Observable` exclusively - alike MVVM pattern). I'm working a very performance-sensitive code where I have full control over inputs but using `ChangeDetectionStrategy.OnPush` fires change detection not just for input changes but also control event fires (like click events), which degrades the performance greatly by re-running unwanted change detection sweep on every click. 

So I opted to detach `ChangeDetectorRef` completely but then now the `AsyncPipe` stops working. Is this expected behavior for `AsyncPipe`? And if so, is there a way to work around this? Or perhaps, a way to make `ChangeDetectorStrategy.OnPush` to not fire on control event fires?

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Fixes https://github.com/angular/angular/issues/12356

### CODE REVIEW

**Comment by DzmitryShylovich** [comment]
Probably need @tbosch feedback on this feature. Are we going to support this?

File: modules/@angular/common/src/pipes/async_pipe.ts [hunk 1]
```diff
@@ -131,10 +133,17 @@ export class AsyncPipe implements OnDestroy {
...
   }
 
-  private _updateLatestValue(async: any, value: Object) {
+  private _updateLatestValue(async: any, value: Object, emitDetached: boolean) {
     if (async === this._obj) {
       this._latestValue = value;
-      this._ref.markForCheck();
+      if (this._ref.isDetached()) {
+        if (emitDetached) {
+          this._ref.reattach();
```
Comment (laco0416) [inline modules/@angular/common/src/pipes/async_pipe.ts]: why not `detectChanges()` ?
